projects/observability/src/shared/dashboard/data/graphql/table/entity/entity-table-data-source.model.ts

```diff
@@ -59,17 +68,38 @@ export class EntityTableDataSourceModel extends TableDataSourceModel {
     filters: GraphQlFilter[],
     request: TableDataRequest<SpecificationBackedTableColumnDef>
   ): GraphQlEntitiesQueryRequest {
+    // tslint:disable-next-line: strict-boolean-expressions
+    if (this.isClientSideRendered) {
+      request.clientSideFilters = request.filters;
+      if (request.sort !== undefined) {
+        request.clientSideSort = {
+          direction: request.sort.direction,
+          column: request.sort.column
+        };
+      }
+    }
+    const offset = this.isClientSideRendered ? 0 : request.position.startIndex;
+
+    let sort = request.sort && {
+      direction: request.sort.direction,
+      key: request.sort.column.specification
+    };
+
+    if (this.isClientSideRendered && this.clientSideSort !== undefined) {
+      sort = {
+        direction: this.clientSideSort.direction,
+        key: request.columns[this.clientSideSort.defaultSortColumnIndex].specification
+      };
+    }
+
     return {
       requestType: ENTITIES_GQL_REQUEST,
       entityType: this.entityType,
       properties: request.columns.map(column => column.specification).concat(...this.additionalSpecifications),
-      limit: this.limit !== undefined ? this.limit : request.position.limit * 2, // Prefetch 2 pages
-      offset: request.position.startIndex,
-      sort: request.sort && {
-        direction: request.sort.direction,
-        key: request.sort.column.specification
-      },
-      filters: [...filters, ...this.toGraphQlFilters(request.filters)],
+      limit: this.limit ?? request.position.limit * 2, // Prefetch 2 pages,
```

Comment:
What's the logic behind 2 pages?

Reply:
That's the default logic that HT folks had, not changing that, for client side filtering, we are explicitly passing the `this.limit` field.